sympy/matrices/tests/test_matrices.py

```diff
@@ -2901,13 +2901,30 @@ def test_atoms():
 def test_pinv():
     # Pseudoinverse of an invertible matrix is the inverse.
     A1 = Matrix([[a, b], [c, d]])
-    assert simplify(A1.pinv()) == simplify(A1.inv())
+    assert simplify(A1.pinv(method="RD")) == simplify(A1.inv())
+
+    # XXX Computing pinv using diagonalization makes an expression that
+    # is too complicated to simplify.
+    # A1 = Matrix([[a, b], [c, d]])
+    # assert simplify(A1.pinv(method="ED")) == simplify(A1.inv())
```

Comment:
I'm only looking for a way to make this commented-out test as some allowed timeout failings.
Things like this is taking too much time, to see it fail
Is there any way to specify the timeout?

Reply:
This is pretty quick. Is it convincing enough?
```python
import random
q=A1.pinv(method="ED")
w=A1.inv()
v=(a,b,c,d)
for do in range(5):
  reps = dict(zip(v, (random.randint(-10**5,10**5) for i in v)))
  assert all(comp(i.n(), j.n()) for i, j in zip(q.subs(reps),w.subs(reps)))
```
If you are nervous about dividing be zero you could check that
`all(i.subs(reps) for i in denoms(q))` before doing the comp.

Reply:
This was the sort of thing I worked with in #8561 .